core/Symbols.cc

```diff
@@ -1171,6 +1171,57 @@ vector<Symbol::RequiredAncestor> Symbol::requiredAncestors(const GlobalState &gs
     return res;
 }
 
+// All required ancestors by this class or module
+vector<Symbol::RequiredAncestor> requiredAncestorsTransitiveInternal(const GlobalState &gs, SymbolRef sym,
+                                                                     std::vector<SymbolRef> seen) {
```

Comment:
Should `seen` be a reference? If not, this is copying the entire vector for each recursive call.